app/admin/user.rb

```diff
@@ -21,6 +21,16 @@ def update
       end
       super
     end
+
+    def destroy
+      # Before destroy, Checking associated
+      user = User.find(params[:id])
```

Comment:
I thought to add custom message. Yes default behaviour work well. 
Removed unused code and use default behaviour.